frontend/src/scenes/setup/ChangePassword.js

```diff
@@ -0,0 +1,72 @@
+import React, { Component } from 'react'
+import { toast } from 'react-toastify'
+import api from '../../lib/api'
+
+export class ChangePassword extends Component {
+    constructor(props) {
+        super(props)
+        this.state = {
+            oldPassword: '',
+            newPassword: '',
+        }
+    }
+
+    handleChange = event => {
+        this.setState({
+            [event.target.name]: event.target.value,
+        })
+    }
+
+    save = async event => {
+        event.preventDefault()
+        try {
+            await api.update('api/user/change_password', {
+                oldPassword: this.state.oldPassword,
+                newPassword: this.state.newPassword,
+            })
+            toast.success('Password changed')
+            // Give the user enough time to see the "Password change" toast
+            setTimeout(() => {
+                window.location.reload()
```

Comment:
What's the reason for this reload?

Reply:
Django automatically invalidates the user's session on a [password change](https://docs.djangoproject.com/en/3.0/topics/auth/default/#changing-passwords). This means that if you click on some other element on the page you will be redirected to the login page. 

So in order for the users to not be surprised because of the above, they instead get redirected after 2 seconds to the login page automatically.

**1. Password change success**
<img width="1149" alt="Screenshot 2020-04-04 at 11 04 16 AM" src="https://user-images.githubusercontent.com/1487006/78423277-8eeea380-7665-11ea-985b-d701f1f3e79e.png">

**2. Redirected to login**
![image](https://user-images.githubusercontent.com/1487006/78423297-b04f8f80-7665-11ea-87ac-55830758a45f.png)

Reply:
That makes sense, but then I think it is worth just using 

```python
update_session_auth_hash(request, form.user)
```

To make sure the user stays logged in, for a better experience!

Reply:
Nice! didn't know that existed.

Reply:
@timgl Done. The user does not get logged out after the password change.